src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

```diff
@@ -170,27 +170,44 @@ def checkSpeedDrainThresholds(self):
         that need updated for speed draining
         """
         enableKeys = []
+        # first, update our summary of condor jobs
+        totalJobs = self.getTotalCondorJobs()
+        if totalJobs is None:
+            msg = "Cannot check speed drain because there was an error fetching job summary from HTCondor."
+            msg += " Will retry again in the next cycle."
+            logging.warning(msg)
+            return []
 
         # get the current speed drain status
         speedDrainConfig = self.agentConfig.get("SpeedDrainConfig")
 
-        # get condor jobs
-        jobs = self.condorAPI.getCondorJobs("", [])
-        if jobs is None:
-            logging.warning("There was an error querying the schedd.  Not checking speed drain thresholds.")
-            return []
-
         # loop through the speed drain configuration and make a list of what thresholds have been hit
         for k, v in viewitems(speedDrainConfig):
             # make sure keys in the speed drain config are valid
             if k in self.validSpeedDrainConfigKeys and isinstance(v['Threshold'], int) and isinstance(v['Enabled'], bool):
                 # we always want to apply the condor priority change if the threshold is hit
                 if not v['Enabled'] or k == 'CondorPriority':
                     logging.info("Checking speed drain threshold for %s. ", k)
-                    if len(jobs) < v['Threshold']:
+                    if totalJobs < v['Threshold']:
                         logging.info("Agent will update speed drain configuration for %s. ", k)
                         enableKeys.append(k)
             else:
                 logging.warning("Speed drain configuration error for %s.  Please check aux db contents.", k)
 
         return enableKeys
+
+    def getTotalCondorJobs(self):
+        """
+        Retrieve a summary of the jobs in condor and return an absolute number
+        of the jobs in Idle and Running states.
+        :return: returns an integer with the total number of jobs, or None if it failed.
+        """
+        jobs = self.condorAPI.getCondorJobsSummary()
+        if not jobs:
+            return None
+
+        results = 0
+        if jobs:
+            for state in self.condorStates:
```

Comment:
Do we care only about `Idle` and `Running` jobs here? What about the rest of the states? If we are about to drain a schedd my understanding is that we should suppose that it is free of not only `Idle` && `Running` jobs but also the ones in `Hold`. And there could be many reasons for a job to be in `Hold`, some of which may even require some debugging.

Reply:
Yes, here we should consider only Idle/Running. Because depending on their total number, we bump their priority or not. So we do not want to affect this decision by jobs lingering around in the `Held` (5) or `Removed` (3) states.
It might be useful though in the drain stats... let me check.